concordium-consensus/src/Concordium/KonsensusV1/TreeState/Types.hs

```diff
@@ -66,6 +72,29 @@ instance Serialize BlockMetadata where
       where
         getUTCPOSIXMicros = posixSecondsToUTCTime . (/ 1_000_000) . realToFrac <$> getWord64be
 
+-- |A class for structures that canonincally include 'BlockMetadata'.
+class HasBlockMetadata bm where
+    -- |Get the block metadata.
+    blockMetadata :: bm -> BlockMetadata
+
+    -- |The height of the block.
+    blockHeight :: bm -> BlockHeight
+    blockHeight = bmHeight . blockMetadata
+    {-# INLINE blockHeight #-}
+
+    -- |The time that the block is received by the consensus layer (i.e. when it is deserialized).
```

Comment:
It's a point in time just **before** deserialization and the block processing kicks off...

Well, it will be looking at: https://github.com/Concordium/concordium-node/tree/multiversion